cli/src/main/java/org/jboss/as/cli/embedded/EmbedHostControllerHandler.java

```diff
@@ -276,7 +276,7 @@ protected void doHandle(CommandContext ctx) throws CommandLineException {
                 do {
                     status = hostController.getProcessState();
 
-                    if (status == null || "starting".equals(status)) {
```

Comment:
I believe status == null is a likely scenario while the host is starting. HostControllerImpl.currentProcessState starts as null and won't change to something else until it receives a notification of a change via ProcessStateNotifier.

Reply:
Yes, I change it because I think we are never going to get a `null` as returned value here if https://github.com/wildfly/wildfly-core/blob/c1c217a784283e1f9413bc08da5bce16dd5061f9/cli/src/main/java/org/jboss/as/cli/embedded/EmbedHostControllerHandler.java#L266 is executed without throwing any exceptions.

The thread which updates the `HostControllerImpl.currentProcessState` via ProcessStateNotifier listener propertyChange() is the "Controller Boot Thread", which is the same thread that calls the done() for the future container. That means https://github.com/wildfly/wildfly-core/blob/c1c217a784283e1f9413bc08da5bce16dd5061f9/embedded/src/main/java/org/wildfly/core/embedded/EmbeddedHostControllerFactory.java#L279 is always returned with the HostControllerImpl.currentProcessState as "starting" and never with null.

This is pretty hidden; it is better to keep the null check and avoid these subtleties to make the code easier to follow. I changed it. Thanks!

Reply:
Thanks; that makes sense. When I wrote that comment I thought it was odd no test hit the race I was concerned about. Now I know why. :)